docs/blueprints/linux-setup.md

```diff
@@ -0,0 +1,181 @@
+# Storage node installation and update on Linux
+
+## Abstract
+
+The design document outlines the steps needed to ease installation and provide auto-update of storage nodes on Linux without the use of Docker.
+
+
+## Background
+
+
+The idea is to have two main components, like for Windows: the storage node binary and a storage node updater that will take care of updating the storage node binary according to the rollout version mechanism.
+
+The parts we need:
+- storagenode as a service
+- a system for updating the storagenode binary aka. the updater (with rollout versioning support)
+- a system for updating the updater 
+- a way to collect the configuration data from the user during the installation
+- packaging to ship the above
+
+Linux systems are not unified as windows is. Hence, we have to make choices that will allow us to cover most linux users, with special regards to the raspbian system.
+
+Most commonly used linux distributions seem to be (there are no reliable stats):
+- debian-based (Debian, Ubuntu, Mint, Kali, Raspbian)
+- Red Hat-based (Fedora)
+- Arch Linux (Manjaro is on the rise according to [distrowatch](https://distrowatch.com/dwres.php?resource=popularity)).
+
+All these distributions are shipped with systemD as a service manager.
+
+
+
+
+## Design
+The installer will be a debian package. We choose to auto-update the binary, even if this will make the package not following debian guidelines
+- When installing the package, it should prompt for user input for:
+  - Wallet address
+  - Email
+  - External address/port
+  - Advertised storage
+  - Identity directory
+  - Storage directory
+- Generate `config.yaml` file with the user configuration.
+
+We choose to reuse the storagenode-updater and the recovery mechanism used in windows. They will be daemonized using systemD. The storagenode updater will auto-update. A recovery will be triggered if the updated updater service fails to restart.
+We will use debconf to retrieve user data. 
+
+The debian package will NOT contain the storagenode and storagenode-updater binaries. They will be downloaded as part of the post-installation script. A separate git repository will be created for holding the debian package.
```

Comment:
Shouldn't it contain the storagenode-updater?

Reply:
No, it is downloaded during the installation process. Do you mean you think it should be? (and in this case, why?)

Reply:
Right, the later. I think it should be. I think the updater should be stable and not be updating itself right? If it is updating itself, then we can just do away with the updater entirely and put that directly in the storage node binary (it can just update itself). It would be nice if the deb at least contained the working binary from the time the pkg was built. It would also be nice if there was a flag in a config that could disable automatic updates (in case the user wants to manage that themselves).

Reply:
We have to handle rollout versioning, that's why we auto-update the binary. About the config flag for disabling it, I'm not sure. We would get the risk of most nodes doing, breaking the rollout process.

For the updater being integrated into the binary, I don't know if it would be best. But as this is the way it is done under windows (with the code already available, if I remember well, only specific commands for stopping and restarting a service under linux need to be integrated), we're doing it the same way.

I have no strong argument against including the binary, except that this would make things more complicated. Should we then update the debian package for each release? The core of the package (apart from the binary of course) would not change, but we would have to update its version too.

Reply:
Integrating updater into storagenode binary is a bad idea since we don't have any recovery and detection of what fails.

For the updater updating itself. Ideally it should only rarely update itself since it's more dangerous. Ideally, yes, it won't ever update itself.

Including the binary in deb. @thepaul might have some good insight here.

Reply:
I think it would be a good idea to have the updater in the deb, and to update the deb every time there is a change to the updater. But our needs are quite different from the normal Debian services that I'm used to thinking about, so I might be missing some key insights.

One reason _not_ to put the updater in the deb is that if we needed to put out an urgent fix, we could be sure of nearly all users getting the fix in a short amount of time (because the updater would update itself, rather than waiting for the user to do an `apt update`). I don't know how likely that situation is, though, and it sort of just shifts the problem: there's still the thing in the deb that downloads the updater; what if we need to push out an urgent fix to _that_?

Reply:
I wouldn't it shifts it, but that it decreases it. Even if we included the updater,we would still face the possibility to have to push an urgent fix to the part that downloads the binary. In the case the script that download the binary(ies) fails, the installation will certainly fail, so I think this risk is "easier" to handle that for example an updater failure. I mean, some people would probably notice this kind of problem faster, just because they cannot have a working installation. The updater, on its side, could fail by many ways unnoticeable by the user.

Reply:
Yes, I agree, it does decrease the scope of the problem while shifting it. I just mean it doesn't eliminate the problem entirely.